src/lua/utils.c

```diff
@@ -800,6 +824,12 @@ tarantool_lua_utils_init(struct lua_State *L)
 	CTID_INTERVAL = luaL_ctypeid(L, "struct interval");
 	assert(CTID_INTERVAL != 0);
 
+	/* Overload os.getenv() with our safe variant. */
+	lua_getglobal(L, "os");
+	lua_pushcfunction(L, luaT_getenv);
+	lua_setfield(L, -2, "getenv");
+	lua_pop(L, 1);
```

Comment:
Personally I prefer to leave an original function available via some appropriate `foo._internal` namespace. It allows a user to switch to the old implementation if he/she experience a problem that we overlooked. Of course, it is not what we'll document, but it may be used as a way to workaround some problem for our customer.

Reply:
Our implementation of `getenv` looks trivial and correct. I can't imagine what can possibly go wrong with it so that the user chooses to switch to the vanilla version. I don't think we should bother.

Reply:
The chance of a problem is actually small, so I'll not insist here. I would let Serge to decide, whether we want to bother (or close this discussion without changes).

Reply:
I'd rather not bother.